ReactCommon/fabric/uimanager/Scheduler.cpp

```diff
@@ -147,26 +147,30 @@ void Scheduler::stopSurface(SurfaceId surfaceId) const {
   SystraceSection s("Scheduler::stopSurface");
 
   long commitStartTime = getTime();
-  shadowTreeRegistry_.visit(
-      surfaceId, [commitStartTime](const ShadowTree &shadowTree) {
-        // As part of stopping the Surface, we have to commit an empty tree.
-        return shadowTree.tryCommit(
-            [&](const SharedRootShadowNode &oldRootShadowNode) {
-              return std::make_shared<RootShadowNode>(
-                  *oldRootShadowNode,
-                  ShadowNodeFragment{
-                      /* .tag = */ ShadowNodeFragment::tagPlaceholder(),
-                      /* .rootTag = */
-                      ShadowNodeFragment::surfaceIdPlaceholder(),
-                      /* .props = */ ShadowNodeFragment::propsPlaceholder(),
-                      /* .eventEmitter = */
-                      ShadowNodeFragment::eventEmitterPlaceholder(),
-                      /* .children = */
-                      ShadowNode::emptySharedShadowNodeSharedList(),
-                  });
-            },
-            commitStartTime);
-      });
+  
+  auto hasShadowTree = shadowTreeRegistry_.visit(
```

Comment:
Early return in case of exception is NOT a valid solution (especially here where stopSurface does not have a way to communicate back about the problem).
The caller side must ensure that the surface with given id exists (so it should not call the method twice).